src/hypertable_restrict_info.c

```diff
@@ -299,18 +388,84 @@ hypertable_restrict_info_add_op_expr(HypertableRestrictInfo *hri, PlannerInfo *r
 							   &lefttype,
 							   &righttype);
 
-	return dimension_restrict_info_add(dri, strategy, c);
+	dimValues = func_get_dim_values(c, use_or);
+	return dimension_restrict_info_add(dri, strategy, dimValues);
+}
+
+static DimensionValues *
+create_dimension_values(List *values, Oid const_type, bool use_or)
+{
+	Oid			base_el_type;
+	DimensionValues *dimValues;
+
+	dimValues = palloc(sizeof(DimensionValues));
+	dimValues->values = values;
+	dimValues->use_or = use_or;
+	/* if an array type, lets get the base element type */
+	base_el_type = get_element_type(const_type);
```

Comment:
I think this logic should be in `dimension_values_create_from_array` and it should be an error if it fails. This kind of code where you assume a reason for a failure is always more brittle. In  `dimension_values_create_from_array` we know it should be an array.